### DISCUSSION TITLE
Spacecraft Command Languages

### DISCUSSION THREAD

**Discussion by jhauerst** [Terrapin-Rocket-Team/SRAD-Avionics#110] [Ideas]
Something that could be used to improve our ability to send commands to the flight computer could be using or implementing our own spacecraft command language. Basically it would allow us to write commands in (somewhat) natural language and translate those to hex codes that are sent and decoded by the flight computer (see Wikipedia article [here](https://en.wikipedia.org/wiki/Spacecraft_command_language)). These languages are what are used to control actual spacecraft, so it would also be good experience to learn how they work.

Basically I'm thinking it would help on the ground station side, since we could just type any command into a text box, while maybe having buttons for commons ones or ones that would need to be sent quickly. Using hex codes could also be a better way of sending commands than sending every configuration option every time we send a command, and since it's already just a number, it would be easy to encode to base 91 for including in the APRS message.

**Comment by mallamacimj** [comment]
I have a lot of opinions on this... I would always hesitate to create your own language as DSLs (domain-specific languages) are notoriously hard to maintain. Also I don't think it is needed because there is a very limited small set of commands that we would actually send. I think defining that list of commands is important to do before (or at least in parallel) to creating the user interface for commanding.
